networkit/cpp/centrality/ComplexPaths.cpp

```diff
@@ -0,0 +1,203 @@
+                if (in != seed
+                    && std::find(moreSeeds.begin(), moreSeeds.end(), in) == moreSeeds.end())
+                    moreSeeds.push_back(in);
+            }
+            sample(moreSeeds.begin(), moreSeeds.end(), std::back_inserter(seeds), needMore,
+                   randomGen);
+            if (seeds.size() < needSeeds)
+                WARN("complexPathsGraph: ", seed, ": not enough indirect neighbors.");
+        }
+    }
+
+    seeds.insert(seeds.begin(), seed); // put in front, so that direct neighbors are activated first
+
+    for (auto u : seeds) {
+        activated[u] = true;
+        if (adopters)
+            adopters->push_back(u);
+        auto neighbors = Graph::NeighborRange<>(g, u);
+
+        for (auto v : neighbors) {
+            addNewEdge(complex_g, u, v);
+        }
+    }
+    // compute influences: degrees of all nodes in complex_g so far
+    auto computeInfluence = [&](node u) { influence[u] = complex_g.degree(u); };
+    complex_g.parallelForNodes(computeInfluence);
+
+    bool spread;
+    do {
+        spread = false; // at start of new iteration
+        complex_g.forNodes([&](node u) {
+            if (influence[u] >= threshold) {
+                if (!activated[u]) {
+                    activated[u] = true;
+                    if (adopters)
+                        adopters->push_back(u);
+                    spread = true;
+                }
+                auto uNeighbors = Graph::NeighborRange<>(g, u);
+                for (auto v : uNeighbors) {
+                    addNewEdge(complex_g, u, v);
+                }
+            }
+        });
+        if (spread) { // new influences in growing complex_g
+            complex_g.parallelForNodes(computeInfluence);
+        }
+    } while (spread);
+
+    return complex_g;
+}
+
+static auto min_max_norm(vector<double> data) {
+    auto [min, max] = std::minmax_element(data.begin(), data.end());
+    auto diff = *max - *min;
+    vector<double> norm(data);
+    // Hint: Use structured bind "min" as init-capture in order to avoid compiler errors
+    // for C++17 and Clang (at least up until C++20).
+    auto scale = [diff, &min = min](auto e) { return (e - *min) / diff; };
+    for (auto &e : norm)
+        e = scale(e);
+
+    return norm;
+}
+
+vector<double> ComplexPathAlgorithm::complexPathLength(count t) {
+    const auto &g = inputGraph;
+    auto n = g.numberOfNodes();
+    vector<double> PLci(n);
+    constexpr auto infDist = std::numeric_limits<edgeweight>::max();
+
+    g.parallelForNodes([&](node u) {
+        auto c_g = complexPathsGraph(u, t, nullptr);
+        auto noPaths = false;
+        NetworKit::BFS bfs(c_g, u, noPaths);
+        bfs.run();
+        auto distances = bfs.getDistances();
+        for (auto &d : distances)
+            if (d == infDist)
+                d = 0.0;
+        auto distanceSum = std::accumulate(distances.begin(), distances.end(), 0.0);
+        PLci[u] = distanceSum / n;
```

Comment:
Division between `double` and `uint64_t`
```suggestion
        PLci[u] = distanceSum / static_cast<double>(n);
```